src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -409,7 +436,7 @@ else if (planKey.toLowerCase().contains("-solution")) { // TODO: transfer this i
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     public ResponseEntity<Result> getLatestResultWithFeedbacks(@PathVariable Long participationId) {
         log.debug("REST request to get latest result for participation : {}", participationId);
-        Participation participation = participationService.findOne(participationId);
+        StudentParticipation participation = participationService.findOne(participationId);
```

Comment:
All endpoints that now explicitly instantiate the StudentParticipation don't work anymore for Template/SolutionParticipations.
Besides new endpoints his will necessitate work in the programming exercise UIs (edit, view, edit in editor) because we have to make explicit requests to the different endpoints.

Reply:
As discussed: It would save us a lot of time if the Result Endpoints would be Participation neutral.
Otherwise a lot of neutral GUI components in our client will have to differentiate the different participation types, e.g. when retrieving the latest result for the updating result component in the code-editor.